src/model/enums/misc.rs

```diff
@@ -0,0 +1,105 @@
+use serde::{Deserialize, Serialize};
+use strum::{AsRefStr, Display, EnumString};
+
+/// disallow: interrupting_playback, pausing, resuming, seeking, skipping_next, skipping_prev, toggling_repeat_context, toggling_shuffle, toggling_repeat_track, transferring_playback
+#[derive(
+    Clone, Serialize, Deserialize, Copy, PartialEq, Eq, Debug, Hash, EnumString, AsRefStr, Display,
```

Comment:
May I ask why we use `AsRefStr` instead of `ToString` for these derive macros?

Reply:
Actually, I can't tell the difference between `EnumString` and `Tostring`, which are almost same, so I choose `EnumString` with additional attribute `#[strum(serialize_all = "snake_case")]`.

`AsRefStr` is another trait to derive, which performs as same as the old function `as_str()`. Thus, instead of prefer `Tostring` to `AsRefStr`, I prefer to keep them both, but I choose `EnumString` instead of `ToString`.

Reply:
The difference is that `EnumString` implements `FromStr`, and `ToString` implents `ToString`. So you can't just choose between these two because they do different things. If anything, the choice is between `ToString`, which returns a `String`, and `AsRefStr`, which returns a `&str`. But IMO the former makes more sense in the context of what we need: to convert something into a string.

But that makes me wonder what these derives are even for. `aspotify` for example doesn't seem to implement `ToString` and `FromStr`. Are they needed for serialization or something? Isn't that done by `serde`?

Reply:
>  Isn't that done by serde?

No, it's nothing related to `serialization`, `to_string()` is for setting something like a query parameter:
```rust
    #[maybe_async]
    pub async fn tracks<'a>(
        &self,
        track_ids: impl IntoIterator<Item = &'a str>,
        market: Option<Country>,
    ) -> ClientResult<FullTracks> {
        // TODO: this can be improved
        let mut ids: Vec<String> = vec![];
        for track_id in track_ids {
            ids.push(self.get_id(Type::Track, track_id));
        }

        let mut params = Query::new();
        if let Some(market) = market {
            params.insert("market".to_owned(), market.to_string());
        }

        let url = format!("tracks/?ids={}", ids.join(","));
        let result = self.get(&url, None, &params).await?;
        self.convert_result(&result)
    }

```
`enums` is not only for model, but also used in somewhere else.

Reply:
> No, it's nothing related to serialization, to_string() is for setting something like a query parameter:

OK, thanks for clearing it up. So wouldn't it make more sense to convert to a string with `.to_string()` in that case instead of `.as_ref()`? AFAIK the latter is mostly used in smart pointers like `Box`.

Reply:
> AFAIK the latter is mostly used in smart pointers like Box. Some of your enums derive both AsRefStr and Display; I think we would be fine with just a single ToString.

Yep, good point, I would replace all `EnumString, AsRefStr, Display` to `ToString`, your point makes sense :)

Reply:
I've been investigating and in fact, we could avoid `strum` altogether and use this from `serde_json` - https://docs.serde.rs/serde_json/ser/fn.to_string.html

Reply:
Great! That should reduce compile times as well.

Reply:
All I've found is this crate - https://docs.rs/serde_plain/0.3.0/serde_plain/fn.to_string.html - but I'm not sure if it's better than using `strum`. It saves us from repeating the lowercase conversions and renames, and it seems to be lighter than `strum`, but you'd have to use `serde_plain::to_string(&val)?` instead of `var.to_string()`.

Reply:
IMHO, this syntax is a little bit weird, perhaps we could find another way to do so.
 
> you'd have to use serde_plain::to_string(&val)? instead of var.to_string().

Reply:
I think it's impossible:

1. `Serialize` is an external trait from `serde`
2. `ToString` is an external trait from `std`

Thus, a crate like `serde_json` can't have a blanket implementation of `ToString` for all `T: Serialize` types because both are external to it. If anything, it could create an equivalent trait for `ToString`, like `OurToString`, but that wouldn't play well with generic types over `ToString`.

Reply:
Yeah, I agree with you, `strum` is a solid solution for our case. By the way, I have pushed some new commits, including a bit model modification. I will keep refactoring model after this PR merged.